File: examples/vbnet/ex_viewportresolution.vb
Imports Rhino

Namespace examples_vb
  <System.Runtime.InteropServices.Guid("4F4C1AE6-725F-44AA-A35D-6D891681BB0B")> _
  Public Class ex_viewportresolution
    Inherits Rhino.Commands.Command
    Public Overrides ReadOnly Property EnglishName() As String
      Get
        Return "vbViewportResolution"
      End Get
    End Property

    Protected Overrides Function RunCommand(doc As RhinoDoc, mode As Rhino.Commands.RunMode) As Rhino.Commands.Result
      Dim avp = doc.Views.ActiveView.ActiveViewport
      RhinoApp.WriteLine([String].Format("Name = {0}: Width = {1}, Height = {2}", avp.Name, avp.Size.Width, avp.Size.Height))
      Return Rhino.Commands.Result.Success
    End Function
  End Class
End Namespace

File: examples/cs/ex_viewportresolution.cs
using Rhino;
using Rhino.Commands;
using System;

namespace examples_cs
{
  [System.Runtime.InteropServices.Guid("75ED2D51-3633-4624-947A-02B15706D3F4")]
  public class ViewportResolutionCommand : Command
  {
    public override string EnglishName { get { return "csViewportResolution"; } }

    protected override Result RunCommand(RhinoDoc doc, RunMode mode)
    {
      var activeViewport = doc.Views.ActiveView.ActiveViewport;
      RhinoApp.WriteLine(String.Format("Name = {0}: Width = {1}, Height = {2}", 
        activeViewport.Name, activeViewport.Size.Width, activeViewport.Size.Height));
      return Result.Success;
    }
  }
}